### ISSUE TITLE
Normalization will not work

### ISSUE THREAD

**Issue by zhangtemplar** [cvqluu/Additive-Margin-Softmax-Loss-Pytorch#3]
I tested it and the normalization will not working:
``` 
       for W in self.fc.parameters():
            W = F.normalize(W, p=2, dim=1)
```

**Comment by gregunz** [comment]
This should fix it:

```python
for _, module in self.fc.named_modules():
    if isinstance(module, nn.Linear):
        module.weight.data = F.normalize(module.weight, p=2, dim=1)
```

**Comment by bnu-wangxun** [comment]
> This should fix it:
> 
> ```python
> for _, module in self.fc.named_modules():
>     if isinstance(module, nn.Linear):
>         module.weight.data = F.normalize(module.weight, p=2, dim=1)
> ```

thanks

**Comment by JisongXie** [comment]
I find this problem, too. I modify it like this:
```
self.fc.weight.data = F.normalize(self.fc.weight.data, p=2, dim=1)
```

**Comment by Charlyo** [comment]
@cvqluu can you fix it in the code? thanks

**Comment by xdn-github** [comment]
I think use weight normalize layer may be a more simlple way:
in line 35 use 
`self.fc = weight_norm(nn.Linear(in_features, out_features, bias=False), name='weight')`
instead of
`self.fc = nn.Linear(in_features, out_features, bias=False)`
and delete
```
       for W in self.fc.parameters():
            W = F.normalize(W, p=2, dim=1)
```
